src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -1373,29 +1402,99 @@ def prepare_inputs_for_generation(
         decoder_input_ids,
         past=None,
         attention_mask=None,
+        decoder_attention_mask=None,
         head_mask=None,
         decoder_head_mask=None,
         cross_attn_head_mask=None,
         use_cache=None,
         encoder_outputs=None,
         **kwargs
     ):
+
         # cut decoder_input_ids if past is used
         if past is not None:
             decoder_input_ids = decoder_input_ids[:, -1:]
 
+        if decoder_attention_mask is not None:  # xla
```

Comment:
nit: From TF-GPT2's `prepare_inputs_for_generation`, maybe the following will be more consistent?

I would guess `decoder_position_ids` would never be passed to `prepare_inputs_for_generation`, so it will be always created here.
(But the question becomes why we even have a such check in TF-GPT2?)

```
        decoder_position_ids = kwargs.get("decoder_position_ids", None)
        ...

        if decoder_position_ids is None:
            if decoder_attention_mask is not None:
                ....
            elif:
                ...
            else:
                ...
```

Reply:
However, see my comment for the (merged) TF-GPT2 PR

https://github.com/huggingface/transformers/pull/17426/files#r889687584

Reply:
I agree with your concerns, but I'm not going to worry about consistency for now :) I haven't managed to get beam search to work, so there is a chance I will have to rewrite all these functions.

After beam search is working then yes, I'd like to revisit these models and make a template for each kind (i.e. decoder-only or encoder-decoder model). Would that work for you, @ydshieh ?

Reply:
Good for me, @gante :-)